### PR TITLE
Airship: Use API key authorization

### PR DESCRIPTION
**Pull request by ujjwal-ab**
## Description of the change

> Using API key for authorization

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by utsabc** [comment]
@ujjwal-ab we tested these changes with the destination also we need to make ui changes as well right?

File: v0/destinations/airship/transform.js [hunk 1]
```diff
@@ -195,19 +176,7 @@ const trackResponseBuilder = async (message, { Config }) => {
...
-        "[Airship] App Key is required for Authentication",
-        400
-      );
-    else
-      throw new CustomError(
-        "[Airship] App Secret is required for authentication",
-        400
-      );
-  }
+  const { apiKey, dataCenter } = Config;
```
Comment (akashrpo) [inline v0/destinations/airship/transform.js:179]: Let's keep `apiKey` validation

File: v0/destinations/airship/transform.js [hunk 2]
```diff
@@ -157,19 +144,13 @@ const trackResponseBuilder = async (message, { Config }) => {
     payload.value.replace(/\s+/g, "_");
   }
   const { appKey, dataCenter, apiKey } = Config;
```
Comment (akashrpo) [inline v0/destinations/airship/transform.js:146]: Let's keep `appKey` validation

File: v0/destinations/airship/transform.js (hunk 1 above)
Comment (ujjwal-ab) [reply v0/destinations/airship/transform.js:192]: Done

File: v0/destinations/airship/transform.js (hunk 2 above)
Comment (ujjwal-ab) [reply v0/destinations/airship/transform.js:153]: Done

File: v0/destinations/airship/transform.js [hunk 3]
```diff
@@ -28,19 +27,14 @@ const {
...
-      throw new CustomError(
-        "[Airship]:: App Secret is required for authentication",
-        400
-      );
-  }
+  const { apiKey, dataCenter } = Config;
+
+  if (!apiKey)
+    throw new CustomError(
+      "[Airship] API Key (Bearer Token) is required for authorization",
```
Comment (akashrpo) [inline v0/destinations/airship/transform.js:34]:
```suggestion
      "[Airship] API Key is required for authorization",
```

Comment (ujjwal-ab) [reply v0/destinations/airship/transform.js]: Done

**Review by akashrpo — DISMISSED** [review]

File: v0/destinations/airship/transform.js [hunk 4]
```diff
@@ -28,19 +27,14 @@ const {
...
-      throw new CustomError(
-        "[Airship]:: App Secret is required for authentication",
-        400
-      );
-  }
+  const { apiKey, dataCenter } = Config;
+
+  if (!apiKey)
+    throw new CustomError(
+      "[Airship] :: API Key is required for authorization",
```
Comment (utsabc) [inline v0/destinations/airship/transform.js:34]:
```suggestion
      "[Airship] :: API Key is required for authorization for Identify events",
```

File: v0/destinations/airship/transform.js [hunk 5]
```diff
@@ -157,19 +151,19 @@ const trackResponseBuilder = async (message, { Config }) => {
     payload.value.replace(/\s+/g, "_");
   }
   const { appKey, dataCenter, apiKey } = Config;
-  if (!apiKey || !appKey) {
+
+  if (!appKey || !apiKey) {
     if (!appKey)
       throw new CustomError(
-        "[Airship] App Key is required for Authentication",
+        "[Airship] App Key is required for authorization",
```
Comment (utsabc) [inline v0/destinations/airship/transform.js:158]:
```suggestion
        "[Airship] App Key is required for authorization for track events",
```

File: v0/destinations/airship/transform.js [hunk 6]
```diff
@@ -157,19 +151,19 @@ const trackResponseBuilder = async (message, { Config }) => {
...
         400
       );
     else
       throw new CustomError(
-        "[Airship] Api Key (Bearer Token) is required for authentication",
+        "[Airship] :: API Key is required for authorization",
```
Comment (utsabc) [inline v0/destinations/airship/transform.js:163]:
```suggestion
        "[Airship] :: API Key is required for authorization for track events",
```

File: v0/destinations/airship/transform.js [hunk 7]
```diff
@@ -195,19 +189,13 @@ const trackResponseBuilder = async (message, { Config }) => {
...
-      throw new CustomError(
-        "[Airship] App Secret is required for authentication",
-        400
-      );
-  }
+  const { apiKey, dataCenter } = Config;
+
+  if (!apiKey)
+    throw new CustomError(
+      "[Airship] :: API Key is required for authorization",
```
Comment (utsabc) [inline v0/destinations/airship/transform.js:196]:
```suggestion
      "[Airship] :: API Key is required for authorization for group events",
```

File: v0/destinations/airship/transform.js (hunk 4 above)
Comment (ujjwal-ab) [reply v0/destinations/airship/transform.js]: Updated

File: v0/destinations/airship/transform.js (hunk 5 above)
Comment (ujjwal-ab) [reply v0/destinations/airship/transform.js]: Updated

File: v0/destinations/airship/transform.js (hunk 6 above)
Comment (ujjwal-ab) [reply v0/destinations/airship/transform.js]: Updated

File: v0/destinations/airship/transform.js (hunk 7 above)
Comment (ujjwal-ab) [reply v0/destinations/airship/transform.js]: Updated

**Review by ItsSudip — APPROVED** [review]

**Review by akashrpo — APPROVED** [review]

**Pull request merged**